crypto/sha/asm/sha512-x86_64.pl

```diff
@@ -2012,7 +2028,18 @@ ()
 my @insns = (&$body,&$body,&$body,&$body);	# 96 instructions
 my $base = "+2*$PUSH8(%rsp)";
 
-	&lea	("%rsp","-$PUSH8(%rsp)")	if (($j%2)==0);
+	if (($j%2)==0) {
+	&lea	("%rsp","-$PUSH8(%rsp)");
+$code.=<<___ if (!$win64);
+.cfi_cfa_expression	%rsp+`$PUSH8-8`,deref,+8
+# copy secondary frame pointer to new location again at -8(%rsp)
+	pushq	$PUSH8-8(%rsp)
+.cfi_cfa_expression	%rsp,deref,+8
+	lea	8(%rsp),%rsp
+.cfi_cfa_expression	%rsp-8,deref,+8
+___
+	}
```

Comment:
here sha256 inner loop this instruction is executed several times with different rsp-frame offset,
therefore the frame pointer needs to be always at -8(%rsp) or 0(%rsp) after the push.
There are absolutely no free registers here.